### PR TITLE
Tempest v2

### PR DESCRIPTION
**Pull request by aewhite**
@sawyercade review, especially the documentation and readme

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,34 +1,93 @@
...
 
-    elasticsearch-<es_version>/bin/plugin -url file:///path/to/tempest/target/releases/tempest-allocator-<version>.zip -install tempest
+TBD
 
-# Configuration (elasticsearch.yml)
+## Configuration
 
-    cluster.routing.allocation.type: com.datarank.tempest.allocator.LocalMinimumShardsAllocator
-(Required) Replaces elasticsearch's default BalancedShardsAllocator behavior with tempest's LocalMinimumShardsAllocator behavior.
+All setting are dynamic are are reloaded for each rebalance request. The defaults should be reasonable for clusters of
```
Comment (sawyercade) [inline README.md]: You have a few extra words here.

File: README.md [hunk 2]
```diff
@@ -1,34 +1,93 @@
...
 
-# Relative Node Balancing
-The allocator attempts to minimize the ratio of the sizes of the most full node to the least full node. That is, it attempts to distribute shards throughtout the cluster such that maxNode.size() / minNode.size() is minimized until it is below the ratio specified by cluster.routing.allocation.probabilistic.range_ratio, or 1.5 by default.
+Tempest is a shard allocator and balancer for Elasticsearch. It is designed to balance by size rather than by shard
+count.
 
-# Elasticsearch Version Support
-Tempest is currently developed against Elasticsearch 1.4.2, but support has been tested up to Elasticsearch 1.7.1. Consequently, tempest can be used with an Elasticsearch cluster of any version in the range 1.4.2-1.7.1, and support may extend further. If there's a specific version you would like tested, please create an issue and we'll verify functionality.
+The default balancer for Elasticsearch attempts to balance a cluster by equally distributing shards. For most use cases
+this is sufficient; However, when custom routing is used, it is possible for some shards to have vastly different sizes.
```
Comment (sawyercade) [inline README.md]: You should add a high-level description here of the balancing algorithm used - searches for N batches of size X, selects the top 10 based on <insert heuristic details here>, repeats until certain heuristic threshold is reached.

Adding this description will help users actually understand what the settings below are actually for, and how they should use them.

File: README.md [hunk 3]
```diff
@@ -1,34 +1,93 @@
...
+* `tempest.balancer.searchQueueSize` - The number of best move chains to consider for final selection (default 10)
+* `tempest.balancer.minimumShardMovementOverhead` - The cost associated with moving a small shard (default 100000000)
+
+The defaults for the settings are pretty good for most cases. Larger clusters (100+ nodes) might benefit from a shallower
+search depth in favor of of a higher search scale factor.
+
+### Limiting
+
+These settings are used to prevent Tempest from being too aggressive with balancing. In particular, they should help prevent
+odd intermediate states that put strain on a single node and prevent costly rebalances that minimum impact on the overall
```
Comment (sawyercade) [inline README.md]: The last part of this sentence is oddly structured and unclear what you're trying to say.

**Comment by kennycason** [comment]
👍 that was fast :)

**Comment by aewhite** [comment]
@sawyercade take another peek.

**Pull request merged**